packages/shared/src/utils/generate-joi.ts

```diff
@@ -179,26 +179,26 @@ export function generateJoi(filter: FieldFilter, options?: JoiOptions): AnySchem
 		}
 
 		if (operator === '_gt') {
-			schema[key] = Number.isSafeInteger(compareValue)
-				? getNumberSchema().greater(compareValue)
+			schema[key] = Number.isSafeInteger(Number(compareValue))
```

Comment:
@rijkvanzanten Is there any possibility where a `numeric` compareValue is not of number type? Tested the following changes on the date comparators and they seem to work as intended, passing the tests. If the casting is required, I think the cast to `Number()` should be added to `_between` and `_nbetween`.

```suggestion
			schema[key] = Number.isSafeInteger(typeof compareValue === 'number' ? Number(compareValue) : compareValue)
```

Reply:
Yes! The system filter interface for example saves the compare value as a string no matter the value, as it can be against any field, so you end up with a 

```
{
  "field": {
    "_eq": "15"
  }
}
```

that in turn caused the logic before to treat "15" as a Date, rather than a number.

I think we should always try the value as a number, and only convert to the date operator when we know it's NaN

Reply:
Pushed a check where the value is set to `NaN` if the value is a date object.